plugins/builtin/target/aws-asg/plugin/aws.go

```diff
@@ -0,0 +1,331 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/hashicorp/nomad-autoscaler/helper/scaleutils"
+)
+
+const (
+	defaultRetryInterval = 10 * time.Second
+	defaultRetryLimit    = 15
+)
+
+// setupAWSClients takes the passed config mapping and instantiates the
+// required AWS service clients.
+func (t *TargetPlugin) setupAWSClients(config map[string]string) error {
+
+	// Load our default AWS config. This handles pulling configuration from
+	// default profiles and environment variables.
+	cfg, err := external.LoadDefaultAWSConfig()
+	if err != nil {
+		return fmt.Errorf("failed to load default AWS config: %v", err)
+	}
+
+	// Check for a configured region and set the value to our internal default
+	// if nothing is found.
+	region, ok := config[configKeyRegion]
+	if !ok {
+		region = configValueRegionDefault
+	}
+
+	// If the default config is empty, update it.
+	if cfg.Region == "" {
+		t.logger.Trace("setting AWS region for client", "region", region)
+		cfg.Region = region
+	}
+
+	// Attempt to pull access credentials for the AWS client from the user
+	// supplied configuration. In order to use these static credentials both
+	// the access key and secret key need to be present; the session token is
+	// optional.
+	keyID, idOK := config[configKeyAccessID]
+	secretKey, keyOK := config[configKeySecretKey]
+	session, _ := config[configKeySessionToken]
+
+	if idOK && keyOK {
```

Comment:
No. I am not aware if the SDK iterates credentials providers and can pull individual pieces out of each to form a single credential set. I had always thought it would try a provider in isolation.